backstage.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: backstage
+  description: |
+    Backstage is an open-source developer portal that puts the developer experience first. 
+  annotations:
+    github.com/project-slug: 'spotify/backstage'
+spec:
+  type: service
```

Comment:
I don't think Backstage should be classified as a service. The main app that people interact with is a `website`, but I guess we could also add another Component, inside the same YAML file, for the backend. Which would be classified as a `service`.

Reply:
It sounds like you're suggesting that the correct solution has a number of different yaml files in the Backstage mono-repo? For example, `packages/app` could be `website` and `packages/backend` could be `service`? This makes sense to me. 

One potentially confusing thing is that the GitHub actions for `spotify/backstage` apply to the whole repo, so which builds would be shown on `/catalog/Component/backstage-frontend`, and which would be shown on `/catalog/Component/backstage-backend`?

Maybe the whole thing is a `library`, since you're supposed to use it to generate websites and services?

I'm starting to see a potential source of friction inside companies who are trying to adopt Backstage! 😂

Reply:
Any thoughts on the name of the YAML file? [The ADR](https://github.com/spotify/backstage/blob/master/docs/architecture-decisions/adr002-default-catalog-file-format.md) doesn't seem to cover that.

Reply:
I will let @shmidt-i or @Rugvip answer how the GitHub Action plugin would handle this. Ideally a single repo should be able to house multiple component definition files in different folders.

To simplify I think using `library` works for now.

Reply:
Perhaps then, more generically `entity.yaml`? Or perhaps `catalog-info.yaml` which is more in line with how we internally name different such info files? @Rugvip @soapraj et al, any more ideas?

Reply:
Was also wondering about naming conventions. I think `component.yaml` and `entity.yaml` might be a bit too generic and run a higher risk of conflicting or being confused with a pre-existing concept. I think `catalog-info.yaml` should be pretty safe and also clear.

Reply:
Especially since it may (although we don't recommend it yet) contain more than one entity, as several yaml documents.

Reply:
@freben Just about to say :grin:, I think that's the only reason we need per-type names internally.

Would also say that `catalog-info.yaml` could be the generic name for descriptors inside a repo that is a general source code repo. In cases where you have dedicated catalog data repos I'd say you can be a lot more flexible with the naming, for example static configuration of all users and groups in a company, resources in an external system, or a protobuf schema repo.

Reply:
Ok. I'll do the following two things:

1. Rename `backstage.yaml` to `catalog-info.yaml`
2. change the `spec.type` to `library`

Does that make sense?

Reply:
> One potentially confusing thing is that the GitHub actions for `spotify/backstage` apply to the whole repo, so which builds would be shown on `/catalog/Component/backstage-frontend`, and which would be shown on `/catalog/Component/backstage-backend`?

This is an interesting point, in a non-monorepo point of view you think that every build is something that changed within the repository, which is a fair assumption, in a monorepo however, I think it's also a fair assumption that you don't really know if this thing changed until you've built it because it depends on so much around it.

I think the current behaviour, although might be verbose, is probably the best way. I think it's good practice to still version everything together in a monorepo, so it makes sense then when a package changes that anything service that references the same `project-slug` should also have a build too.

Reply:
@benjdlambert doesn't that slightly dilute the purpose of Backstage though? Imagine a monorepo with 2000 microservices in it (which Monzo actually have, based on a podcast interview with a staff engineer there). Every Backstage catalog page would just show the same list of thousands of builds, regardless of whether or not the entity you are viewing was affected or not.

Reply:
I'm not saying theres not a use case for filtering the builds based on files changed ;)